### ISSUE TITLE
bug:[victoria-metrics-k8-stack]

### ISSUE THREAD

**Issue by mdesouky** [VictoriaMetrics/helm-charts#1545]
**Chart name and version**
chart: victoria-metrics-k8-stack
version: v0.24.2

**Describe the bug**
When deploying vmalert using `victoria-metrics-k8-stack` chart and configuring notifier to use Oauth2, vmalert pods are unable to start due to not being able to initialize the notifier as in the following logs:
```
2024-10-02T04:32:58.980Z	info	VictoriaMetrics/lib/logger/flag.go:12	build version: vmalert-20240425-145844-tags-v1.101.0-0-g5334f0c2c
2024-10-02T04:32:58.980Z	info	VictoriaMetrics/lib/logger/flag.go:13	command-line flags
2024-10-02T04:32:58.980Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -datasource.basicAuth.passwordFile="secret"
2024-10-02T04:32:58.980Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -datasource.basicAuth.username="vm"
2024-10-02T04:32:58.980Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -datasource.url="secret"
2024-10-02T04:32:58.980Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -evaluationInterval="15s"
2024-10-02T04:32:58.980Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -external.url="https://vmalert.onuptick.com"
2024-10-02T04:32:58.980Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -httpListenAddr=":8080"
2024-10-02T04:32:58.980Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -notifier.oauth2.clientID="209145867298-uj0108mmq5jtgb716oddjr126q86ijj1.apps.googleusercontent.com"
2024-10-02T04:32:58.980Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -notifier.oauth2.clientSecretFile="secret"
2024-10-02T04:32:58.980Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -notifier.oauth2.scopes="openid"
2024-10-02T04:32:58.980Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -notifier.oauth2.tokenUrl="secret"
2024-10-02T04:32:58.980Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -notifier.url="secret"
2024-10-02T04:32:58.980Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -remoteRead.basicAuth.passwordFile="secret"
2024-10-02T04:32:58.980Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -remoteRead.basicAuth.username="vm"
2024-10-02T04:32:58.980Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -remoteRead.url="secret"
2024-10-02T04:32:58.980Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -remoteWrite.basicAuth.passwordFile="secret"
2024-10-02T04:32:58.980Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -remoteWrite.basicAuth.username="vm"
2024-10-02T04:32:58.980Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -remoteWrite.concurrency="4"
2024-10-02T04:32:58.980Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -remoteWrite.disablePathAppend="true"
2024-10-02T04:32:58.980Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -remoteWrite.maxBatchSize="5000"
2024-10-02T04:32:58.980Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -remoteWrite.maxQueueSize="100000"
2024-10-02T04:32:58.981Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -remoteWrite.url="secret"
2024-10-02T04:32:58.981Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -rule="/etc/vmalert/config/vm-victoria-metrics-k8s-stack-rulefiles-0/*.yaml"
2024-10-02T04:32:58.981Z	fatal	VictoriaMetrics/app/vmalert/main.go:162	failed to init: failed to init notifier: failed to create notifier from flag values: failed to configure auth: cannot initialize oauth2: ClientSecret and ClientSecretFile cannot be set simultaneously
```

**Custom values**
Please provide only custom values (excluding default ones):
```yaml
vmalert:
  spec:
    notifiers:
    - oauth2:
        client_id:
          secret:
            key: clientId
            name: google-oidc-secret
        client_secret:
          key: clientSecret
          name: google-oidc-secret
        client_secret_file: null
        scopes:
        - openid
        token_url: https://oauth2.googleapis.com/token
      url: https://alertmanager.onuptick.com
```

VMalert object
```yaml
apiVersion: operator.victoriametrics.com/v1beta1
kind: VMAlert
metadata:
  annotations:
    meta.helm.sh/release-name: victoria-metrics-k8s-stack
    meta.helm.sh/release-namespace: monitor
    operator.victoriametrics/last-applied-spec: '{"image":{"tag":"v1.101.0"},"replicaCount":2,"resources":{"limits":{"cpu":"800m","memory":"2Gi"},"requests":{"cpu":"500m","memory":"1Gi"}},"affinity":{"nodeAffinity":{"requiredDuringSchedulingIgnoredDuringExecution":{"nodeSelectorTerms":[{"matchExpressions":[{"key":"karpenter.sh/nodepool","operator":"In","values":["default"]}]}]}}},"tolerations":[{"key":"dedicated","operator":"Equal","value":"system","effect":"NoSchedule"}],"evaluationInterval":"15s","selectAllByDefault":true,"notifiers":[{"url":"http://vmalertmanager-victoria-metrics-k8s-stack.monitor.svc.cluster.local:9093"}],"remoteWrite":{"url":"https://victoriametrics.onuptick.com/api/v1/write","concurrency":4,"maxBatchSize":5000,"maxQueueSize":100000,"basicAuth":{"username":{"name":"victoria-metrics-auth","key":"username"},"password":{"name":"victoria-metrics-auth","key":"password"}}},"remoteRead":{"url":"https://victoriametrics.onuptick.com","basicAuth":{"username":{"name":"victoria-metrics-auth","key":"username"},"password":{"name":"victoria-metrics-auth","key":"password"}}},"datasource":{"url":"http://victoria-metrics-victoria-metrics-single-server.monitor.svc.cluster.local:8428","basicAuth":{"username":{"name":"victoria-metrics-auth","key":"username"},"password":{"name":"victoria-metrics-auth","key":"password"}}},"extraArgs":{"external.url":"https://vmalert.onuptick.com","remoteWrite.disablePathAppend":"true"}}'
  creationTimestamp: "2024-07-25T04:52:39Z"
  finalizers:
  - apps.victoriametrics.com/finalizer
  generation: 29
  labels:
    app.kubernetes.io/instance: victoria-metrics-k8s-stack
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: victoria-metrics-k8s-stack
    app.kubernetes.io/version: v1.101.0
    helm.sh/chart: victoria-metrics-k8s-stack-0.24.2
  name: victoria-metrics-k8s-stack
  namespace: monitor
  resourceVersion: "82601782"
  uid: 9e599c6e-685d-4977-93cf-9e6db86d8539
spec:
  affinity:
    nodeAffinity:
      requiredDuringSchedulingIgnoredDuringExecution:
        nodeSelectorTerms:
        - matchExpressions:
          - key: karpenter.sh/nodepool
            operator: In
            values:
            - default
  datasource:
    basicAuth:
      password:
        key: password
        name: victoria-metrics-auth
      username:
        key: username
        name: victoria-metrics-auth
    url: http://victoria-metrics-victoria-metrics-single-server.monitor.svc.cluster.local:8428
  evaluationInterval: 15s
  externalLabels: {}
  extraArgs:
    external.url: https://vmalert.onuptick.com
    remoteWrite.disablePathAppend: "true"
  image:
    tag: v1.101.0
  notifiers:
  - oauth2:
      client_id:
        secret:
          key: clientId
          name: google-oidc-secret
      client_secret:
        key: clientSecret
        name: google-oidc-secret
      scopes:
      - openid
      token_url: https://oauth2.googleapis.com/token
    url: https://alertmanager.onuptick.com
  remoteRead:
    basicAuth:
      password:
        key: password
        name: victoria-metrics-auth
      username:
        key: username
        name: victoria-metrics-auth
    url: https://victoriametrics.onuptick.com
  remoteWrite:
    basicAuth:
      password:
        key: password
        name: victoria-metrics-auth
      username:
        key: username
        name: victoria-metrics-auth
    concurrency: 4
    maxBatchSize: 5000
    maxQueueSize: 100000
    url: https://victoriametrics.onuptick.com/api/v1/write
  replicaCount: 2
  resources:
    limits:
      cpu: 800m
      memory: 2Gi
    requests:
      cpu: 500m
      memory: 1Gi
  selectAllByDefault: true
  tolerations:
  - effect: NoSchedule
    key: dedicated
    operator: Equal
    value: system
status:
  reason: 'origin_Err=cannot wait for deployment to become ready: context deadline
    exceeded,podPhase="Running",conditions=name="PodReadyToStartContainers",status="True",message="",name="Initialized",status="True",message="",name="Ready",status="False",message="containers
    with unready status: [vmalert]",name="ContainersReady",status="False",message="containers
    with unready status: [vmalert]",name="PodScheduled",status="True",message="",name="config-reloader",is_ready=true,restart_count=0,state=running
    since: 2024-10-02 04:32:58 +0000 UTC,name="vmalert",is_ready=false,restart_count=5,state=waiting
    with reason="CrashLoopBackOff"'
  updateStatus: expanding
```

**Comment by AndrewChubatiuk** [comment]
hey @mdesouky 
please try to set `client_secret_file: ""` instead of `client_secret_file: null`

**Comment by AndrewChubatiuk** [comment]
it's supposed to be an operator issue. will try to reproduce it

**Comment by AndrewChubatiuk** [comment]
this is a vmalert issue, which was fixed in [1.102.0](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/6471). please either upgrade vmalert image
